### PR TITLE
[Zero-Dim] Support paddle.all/any/min/prod/logsumexp/amax/amin/quantile/some loss output 0D, test=allcase

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

card-66984

支持以下API输出0D：
```
1. paddle.all
2. paddle.any
3. paddle.min
4. paddle.prod
5. paddle.logsumexp
6. paddle.amax
7. paddle.amin
8. paddle.quantile
9. binary_cross_entropy
10. binary_cross_entropy_with_logits
11. margin_ranking_loss
12. l1_loss
13. cross_entropy
14. sigmoid_focal_loss
```

同时将一些 `loss.numpy()[0]` 改成兼容态的 `float(loss)` 形式，在[1]和[]下均能运行。

### CODE REVIEW

**Review by cxxly — APPROVED** [review]
LGTM for composite_backward_api.h

**Review by jeff41404 — APPROVED** [review]
LGTM
